src/index.js

```diff
@@ -1,390 +1,586 @@
-export function h(name, attributes) {
-  var rest = []
-  var children = []
-  var length = arguments.length
+var DEFAULT = 0
+var RECYCLED_NODE = 1
+var TEXT_NODE = 2
 
-  while (length-- > 2) rest.push(arguments[length])
+var XLINK_NS = "http://www.w3.org/1999/xlink"
+var SVG_NS = "http://www.w3.org/2000/svg"
 
-  while (rest.length) {
-    var node = rest.pop()
-    if (node && node.pop) {
-      for (length = node.length; length--; ) {
-        rest.push(node[length])
+var EMPTY_OBJECT = {}
+var EMPTY_ARRAY = []
+
+var map = EMPTY_ARRAY.map
+var isArray = Array.isArray
+
+var merge = function(a, b) {
+  var target = {}
+
+  for (var i in a) target[i] = a[i]
+  for (var i in b) target[i] = b[i]
+
+  return target
+}
+
+var resolved = typeof Promise === "function" && Promise.resolve()
```

Comment:
So now Promises are used in core ? (if supported)

Reply:
Nope, this is just a trick to call your callback as quickly as possible.

Reply:
Why still use `var` in 2019? Please use `const`. No need to be ES3/5 compliant

Reply:
@kristianmandrup Check out CONTRIBUTION.md—it's written there.